model/textparse/protobufparse.go

```diff
@@ -480,3 +508,18 @@ func isNativeHistogram(h *dto.Histogram) bool {
 		h.GetZeroCount() > 0 ||
 		h.GetZeroThreshold() > 0
 }
+
+// isFloatHistogram checks that processed histogram is a float histogram if it
+// some of the histogram defined with float values
+func isFloatHistogram(h *dto.Histogram) bool {
+	floatBuckets := false
+	for _, b := range h.Bucket {
+		if b.CumulativeCountFloat > 0 {
+			floatBuckets = true
+			break
+		}
+	}
+	return h.GetSampleCountFloat() > 0 ||
+		h.GetZeroCountFloat() > 0 ||
+		floatBuckets
```

Comment:
We should check sample count and zero count before the buckets, because this is cheaper and we can exit early in most cases. But if what I said in the other comment about checking length of some slices is true, then this comment is irrelevant.